pkg/gitfs/gitfs.go

```diff
@@ -3,13 +3,14 @@ package gitfs
 import (
 	"fmt"
 	"path"
+	"path/filepath"
 	"strings"
 
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing/object"
 	"github.com/go-git/go-git/v5/plumbing/storer"
 	"github.com/go-git/go-git/v5/storage/memory"
-	"sigs.k8s.io/kustomize/pkg/fs"
+	fs "sigs.k8s.io/kustomize/kyaml/filesys"
```

Comment:
`sigs.k8s.io/kustomize/pkg/fs` package is deprecate in favour of `sigs.k8s.io/kustomize/kyaml/filesys`

Ref: https://github.com/kubernetes-sigs/kustomize/blob/afe77936761f8f7ee735a67f167a8382c8406f74/api/filesys/filesys.go#L8-L9